openhands/runtime/base.py

```diff
@@ -213,6 +213,45 @@ async def on_event(self, event: Event) -> None:
             source = event.source if event.source else EventSource.AGENT
             self.event_stream.add_event(observation, source)  # type: ignore[arg-type]
 
+    def clone_repo(self, github_token: str | None, selected_repository: str | None):
+        if not github_token or not selected_repository:
+            return
+        url = f'https://{github_token}@github.com/{selected_repository}.git'
+        dir_name = selected_repository.split('/')[1]
+        action = CmdRunAction(
+            command=f'git clone {url} {dir_name} ; cd {dir_name} ; git checkout -b openhands-workspace'
+        )
+        self.log('info', 'Cloning repo: {selected_repository}')
+        self.run_action(action)
+
+    def get_custom_microagents(self, selected_repository: str | None) -> list[str]:
+        custom_microagents_content = []
+        custom_microagents_dir = Path('.openhands') / 'microagents'
+
+        files = []
+        if selected_repository:
+            oh_instructions_header = '---\nname: openhands_instructions\nagent: CodeActAgent\ntriggers:\n- ""\n---\n'
+            openhands_instructions = (
+                oh_instructions_header
+                + self.read(FileReadAction(path='.openhands_instructions')).content
+            )
+            custom_microagents_content.append(openhands_instructions)
+
+            dir_name = Path(selected_repository.split('/')[1])
+            files = self.list_files(str(dir_name / custom_microagents_dir))
```

Comment:
this logic should be the same for both repo/no-repo. Basically I think you just want to set `dir_name` differently for the two different cases

You'll also need some error handling here--if `.openhands_instructions` doesn't exist, we'll get (I think) a FileNotFoundError or similar back, which we can safely ignore